### ISSUE TITLE
make with errors on centos 7.8

### ISSUE THREAD

**Issue by lerouxje** [rigtorp/udpreplay#26]
$ cmake --version
cmake version 3.18.2
CMake suite maintained and supported by Kitware (kitware.com/cmake).
$ make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ mkdir build && cd build
$ cmake3 ..
-- The CXX compiler identification is GNU 10.2.0
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /TRAVAIL/udpreplay/build
$ make
Scanning dependencies of target udpreplay
[ 50%] Building CXX object CMakeFiles/udpreplay.dir/src/udpreplay.cpp.o
/TRAVAIL/udpreplay/src/udpreplay.cpp: In function 'int main(int, char**)':
/TRAVAIL/udpreplay/src/udpreplay.cpp:225:32: error: 'const struct udphdr' has no member named 'uh_ulen'
  225 |       ssize_t len = ntohs(udp->uh_ulen) - 8;
      |                                ^~~~~~~
/TRAVAIL/udpreplay/src/udpreplay.cpp:232:28: error: 'const struct udphdr' has no member named 'uh_dport'
  232 |       addr.sin_port = udp->uh_dport;
      |                            ^~~~~~~~
make[2]: *** [CMakeFiles/udpreplay.dir/build.make:83: CMakeFiles/udpreplay.dir/src/udpreplay.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/udpreplay.dir/all] Error 2
make: *** [Makefile:161: all] Error 2

I have the same issue on a redhat7 system..

Can you help me please

**Comment by rigtorp** [comment]
Is this some custom compiler you are using?
```
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++ - works
```

**Comment by ddrinka** [comment]
I have the same problem with GCC 4.8.2.

See here for another project with similar errors:
https://redmine.named-data.net/issues/4852

**Comment by rigtorp** [comment]
Try this fix https://github.com/rigtorp/udpreplay/pull/27

**Comment by lerouxje** [comment]
Hello Erik,

It’s ok with your fix   and gcc/7.3.0  ou more:

[lerouxje@tdell427 buildfix]$ module list

Currently Loaded Modulefiles:

1) gcc/7.3.0

[lerouxje@tdell427 udpreplay]$ mkdir buildfix && cd buildfix

[lerouxje@tdell427 buildfix]$ cp ../src/udpreplay.cpp_fix ../src/udpreplay.cpp

cp : voulez-vous écraser « ../src/udpreplay.cpp » ? y

[lerouxje@tdell427 buildfix]$ echo CMAKE_INSTALL_PREFIX:PATH=/TRAVAIL/opt/udpreplay >> CMakeCache.txt

[lerouxje@tdell427 buildfix]$ cmake  ../

-- The CXX compiler identification is GNU 7.3.0

-- Check for working CXX compiler: /opt/gcc/7.3.0/gcc-7.3.0/bin/g++

-- Check for working CXX compiler: /opt/gcc/7.3.0/gcc-7.3.0/bin/g++ -- works

-- Detecting CXX compiler ABI info

-- Detecting CXX compiler ABI info - done

-- Detecting CXX compile features

-- Detecting CXX compile features - done

-- Configuring done

-- Generating done

-- Build files have been written to: /TRAVAIL/udpreplay/buildfix

[lerouxje@tdell427 buildfix]$ make -j 4

Scanning dependencies of target udpreplay

[ 50%] Building CXX object CMakeFiles/udpreplay.dir/src/udpreplay.cpp.o

[100%] Linking CXX executable udpreplay

[100%] Built target udpreplay



Thank’s a lot



De : Erik Rigtorp <notifications@github.com>
Envoyé : lundi 16 novembre 2020 13:03
À : rigtorp/udpreplay <udpreplay@noreply.github.com>
Cc : Jean-Yves LE ROUX <jean-yves.le_roux@ensta-bretagne.org>; Author <author@noreply.github.com>
Objet : Re: [rigtorp/udpreplay] make with errors on centos 7.8 (#26)



Try this fix #27<https://github.com/rigtorp/udpreplay/pull/27>
